### DISCUSSION TITLE
Discuss and share your setup here

### DISCUSSION THREAD

**Discussion by fdev31** [hyprland-community/pyprland#46] [General]
Question, suggestions & misc topics to discuss

**Comment by markoonakic** [comment]
Im trying to use a scratchpad but i want it to be fullscreen, specifficaly neovim. Currently i have a scratchpad that runs "kitty --class neovim nvim". When i set the class neovim to have the fullscreen window rule the scratchpad starts in fullscreen but instantly goes back to being tiled. 

I think using class + class_match wouldn't work since neovim runs in the kitty terminal and that would impact all other instances of the terminal as well. If you have any ideas or suggestions on how i could achieve this if it is possible let me know.

Thanks in advance!

**Reply by fdev31** [reply]
Interesting, I think this is a common use-case, I'm personally using `toggle_special` plugin for that matter but maybe something hybrid can be interesting, I'll think about that.

About your problem, I suspect something: you are using rules in hyprland.conf.
Try not having anything about this "neovim" class, and rely on the `size` in hyprland.conf instead.

**Comment by markoonakic** [comment]
Can't seem to make the window keep the fullscreen option, tried using a special workspace and the `toggle_special` plugin but wont retain fullscreen that way either. When trying with a `scratchpad` i can at least switch the window to fullscreen manually when it is displayed but when the scratchpad is hidden than brought back again it reverts to not being fullscreen. Also tried messing with `size` and `position` but don't think that will work since the goal is for the window the be fullscreen which means also covering the bar im using.

Maybe it would be a cool option to have to be able to set the scratchpad to fullscreen. If maybe you think of a way to do this some other way let me know :)

**Reply by fdev31** [reply]
Hm, I don't see why size="100% 100%" wouldn't do the trick, there is no special handling of bars etc...

**Comment by markoonakic** [comment]
Managed to do it by setting size="100% 100%" and position="0% 0%" and configuring waybar to be at the bottom layer so that the window would overlap it, thanks for the advice!

**Reply by fdev31** [reply]
Is position really needed ?

**Reply by markoonakic** [reply]
I though it was but checked again and seems to be working without position, sorry for the bother.

**Reply by marcelmanz** [reply]
For me in this case it is needed:
```
[scratchpads.whats]
animation = ""
command = "wasistlos %u"
class = "wasistlos"
size = "100% 100%"
position="0% 0%"
lazy = true
```

If not, I get half the app out of the screen.

**Reply by fdev31** [reply]
If someone wants to look into it ad see why this happens... I'm so lazy for it :p but you can open a bug eventually... (is it still happening, if yes I should probably mention it in the documentation, you all have access to PRs by the way ;p)

**Comment by andre-brandao** [comment]
Hey, I wanted to add some more information to the wiki but could not find out how  to edit or pull request it,


would be useful to add this to scracthpad `process_tracking` documentation, I would add it on line 234


You can also use some websites
```toml
[scratchpads.gpt]
animation = "fromTop"
command = "brave --profile-directory=Default --app=https://chat.openai.com"
class = "brave-chat.openai.com__-Default"
size = "75% 60%"
process_tracking = false
```
** tip
```bash
#you can use rofi to get the name of the class
rofi -show window
```

**Reply by fdev31** [reply]
https://github.com/hyprland-community/pyprland/wiki/scratchpads#process_tracking-use-only-if-really-needed

I used your example and a modified version tip... looks good ?

**Reply by andre-brandao** [reply]
Looks good!! but why not keep both examples?

**Reply by fdev31** [reply]
Added it back !

**Comment by besseddrest** [comment]
## Overlap w/ Hyprland's special workspaces?

First, thanks for all the hard work and providing these extensions!

Curious about Expose as I couldn't figure it out at first - then i realize its a special workspace and seems to clash with existing special workspaces from Hyprland - so it would seem like one should either choose to use scratchpads from Pyprland vs special workspaces in Hyprland - one or the other. Is this the case?

One thing I think could be great with Expose is to give the user a way out, as in i can return to the previous workspace, instead of committing to the Expose active window - just curious if this is something that has been considered/something to expect in the future?

**Reply by fdev31** [reply]
With the provided examples you have a "way back" if you don't change the selected window isn't it ? Is that the behavior you would like ? It feels a bit overkill to add a command just for that but why not... 
Maybe if the idea gets some :+1:  or so...

**Comment by fdev31** [comment]
By the way, I have now a way to export a part of my dotfiles with most of my hyprland related config & some scripts:

https://github.com/fdev31/dotfiles
